go/libkb/passphrase_helper.go

```diff
@@ -134,3 +143,58 @@ func DefaultPassphraseArg(allowSecretStore bool) keybase1.GUIEntryArg {
 		},
 	}
 }
+
+// PassphraseChecker is an interface for checking the format of a
+// passphrase. Returns nil if the format is ok, or a descriptive
+// hint otherwise.
+type PassphraseChecker interface {
+	Check(*GlobalContext, string) error
+}
+
+// CheckerWrapper wraps a Checker type to make it conform to the
+// PassphraseChecker interface.
+type CheckerWrapper struct {
+	checker Checker
+}
+
+// Check s using checker, respond with checker.Hint if check
+// fails.
+func (w *CheckerWrapper) Check(_ *GlobalContext, s string) error {
+	if w.checker.F(s) {
+		return nil
+	}
+	return errors.New(w.checker.Hint)
+}
+
+// PaperChecker implements PassphraseChecker for paper keys.
+type PaperChecker struct{}
+
+// Check a paper key format.  Will return a detailed error message
+// specific to the problems found in s.
+func (p *PaperChecker) Check(g *GlobalContext, s string) error {
+	phrase := NewPaperKeyPhrase(s)
+
+	// check for invalid words
+	invalids := phrase.InvalidWords()
+	if len(invalids) > 0 {
+		g.Log.Debug("paper phrase has invalid word(s) in it")
+		if len(invalids) > 1 {
+			return fmt.Errorf("Please try again. These words are invalid: %s", strings.Join(invalids, ", "))
+		}
+		return fmt.Errorf("Please try again. This word is invalid: %s", invalids[0])
+	}
+
+	// check version
+	version, err := phrase.Version()
+	if err != nil {
+		g.Log.Debug("error getting paper key version: %s", err)
+		// despite the error, just tell the user there was a typo:
+		return errors.New("It looks like there was a typo in the paper key. Please try again.")
```

Comment:
I think that's the `version != PaperKeyVersion` check below, as opposed to this `err != nil` check?